client/models/model_editor.js

```diff
@@ -138,27 +138,55 @@ class Models {
         this.selected_type = 'model';
         this.selected_model = model;
 
+        this.textureSelect();
         this.resetUI();
-    }
 
+    }
+    
     loadAsset(id) {
         if (!id) return;
         if (this.selected_model == id) return;
-
+        
         let pack = this.loaded_asset_pack;
-
+        
         this.selected_type = 'asset';
         this.selected_model = id;
         this.asset_definition = {
             model: 'OBJ/' + id,
             texture: Object.keys(this.asset_packs[pack].textures)[0]
         }
-
+        
         this.needs_resize = true;
-
+        
+        this.textureSelect();
         this.resetUI();
     }
 
+    textureSelect(){
+        let select = document.getElementById("model-dialog-controls-texture");
+        select.options.length = 0;
+
+        let pack = this.loaded_asset_pack;
+
+        this.selected_type === 'asset'
+            ? createList(select, this.asset_packs[pack].textures)
+            : createList(select, WORKSPACES.current_map.loadedArgs.model_textures);
+    }
+
+    textureChange(){
+        let selectVal = document.getElementById("model-dialog-controls-texture").value;
+        
+        let original = this.getCurrentModel();
+
+        this.selected_type === 'asset'
+            ? original.texture = selectVal
+            : original.texture = "models/shared-textures/" + selectVal;
```

Comment:
a) Don't modify the current model, change the 'local_changes'.
b) I think it should be 'this.local_changes.sharedTexture = selectVal'